### PR TITLE
Added EXTERNL definition in netcdf_par.h

### LINKED ISSUE #1507: Failure building with mpicc in current master
Appears to be an issue with `EXTERNL` not being defined properly in `netcdf_par.h`.  Fixing now, should be pretty straight-forward.

### PR DESCRIPTION
**Pull request by WardF**
* fixes #1507

### CODE REVIEW

File: include/nc_tests.h [hunk 1]
```diff
@@ -17,11 +17,12 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
+#include "netcdf.h"
 #include "nc_logging.h"
 #ifdef USE_PARALLEL
 #include "netcdf_par.h"
```
Comment (edhartnett) [inline include/nc_tests.h]: Might be better to have netcdf_par.h #include netcdf.h?

**Comment by DennisHeimbigner** [comment]
Why should netcdf_par.h be different than e.g. netcdf_mem.h?

**Comment by WardF** [comment]
Either solution would work; in this case, I’m inclined to err on the side of verbosity.  Trying to untangle what has been included, and from where, can be a headache.  Or perhaps it’s just me :).  I am attracted to the idea that you must include “netcdf.h” if you are working with the ‘default’ NetCDF API.

**Comment by edwardhartnett** [comment]
Well if you must include netcdf.h when you use netcdf_par.h or netcdf_mem.h, then why not have those files do the work?

The reason it's better is to prevent issues exactly like the one that occurred here.

**Comment by DennisHeimbigner** [comment]
The rule as added to the include/netcdf_XXX.h files is this:
/*
 * In order to use any of the netcdf_XXX.h files, it is necessary
 * to include netcdf.h followed by any netcdf_XXX.h files.
 * Various things (like EXTERNL) are defined in netcdf.h
 * to make them available for use by the netcdf_XXX.h files.
*/

**Comment by edwardhartnett** [comment]
You are telling people to do something we can do. No one reads the documentation, as you know! ;-)

**Pull request merged**